### PR TITLE
Update forward declarations of `rcl_lifecycle` types

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rcl/pull/938.

### CODE REVIEW

**Comment by hidmic** [comment]
CI above `rcl_lifecycle` (including `rclcpp_lifecycle`):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15319/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10030/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12999/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15512/)

File: rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp [hunk 1]
```diff
@@ -21,8 +21,8 @@
 
 #include "rcutils/allocator.h"
 
-// forward declare rcl_state_t
-typedef struct rcl_lifecycle_state_t rcl_lifecycle_state_t;
+// forward declare rcl_lifecycle_state_t
+typedef struct rcl_lifecycle_state_s rcl_lifecycle_state_t;
```
Comment (clalancette) [inline rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp:25]: Can we just remove this completely and `#include <rcl_lifecycle/data_types.h>`?

File: rclcpp_lifecycle/include/rclcpp_lifecycle/transition.hpp [hunk 2]
```diff
@@ -22,8 +22,8 @@
 
 #include "rcutils/allocator.h"
 
-// forward declare rcl_transition_t
-typedef struct rcl_lifecycle_transition_t rcl_lifecycle_transition_t;
+// forward declare rcl_lifecycle_transition_t
+typedef struct rcl_lifecycle_transition_s rcl_lifecycle_transition_t;
```
Comment (clalancette) [inline rclcpp_lifecycle/include/rclcpp_lifecycle/transition.hpp:26]: Same here?

File: rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp (hunk 1 above)
Comment (hidmic) [reply rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp]: Sure. See https://github.com/ros2/rclcpp/pull/1788/commits/d1fb7db56e8e6cc43c9f0a9cd675997bee8aa690.

File: rclcpp_lifecycle/include/rclcpp_lifecycle/transition.hpp (hunk 2 above)
Comment (hidmic) [reply rclcpp_lifecycle/include/rclcpp_lifecycle/transition.hpp]: Same as above.

**Review by clalancette — APPROVED** [review]

**Comment by christophebedard** [comment]
Nightlies + source builds are broken because https://github.com/ros2/rcl/pull/938 was merged but this PR wasn't, e.g. https://ci.ros2.org/view/nightly/job/nightly_linux_release/2053/console#console-section-8

**Comment by clalancette** [comment]
> Nightlies + source builds are broken because [ros2/rcl#938](https://github.com/ros2/rcl/pull/938) was merged but this PR wasn't, e.g. https://ci.ros2.org/view/nightly/job/nightly_linux_release/2053/console#console-section-8

Ug, good call.  This seems to have built for me locally with this patch in place, so I'm going to go ahead and merge it without further CI; it can't be worse than the current state.

**Comment by hidmic** [comment]
You beat me to it @clalancette. Sorry for the fallout :man_facepalming:

**Pull request merged**